File: CodeForces Contest Solutions/Codeforces_Round_1016 _(Div. 3)/B_Expensive_Number.cpp
#include <iostream>
using namespace std;
#define ll long long
int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        string n;
        cin >> n;
        ll ans = 0, digit = 0, zero = 0;
        bool haveZero=false, nonZero=false;
        for (int i = n.size() - 1; i >= 0; i--)
        {
            int ld = n[i] - '0';

            if (ld == 0)
                haveZero = true;

            if (ld == 0 && !nonZero)
            {
                zero++;
            }
            else if (ld != 0)
            {
                ans++;
                nonZero = true;
            }

            digit++;
        }

        if (haveZero)
        {
            cout << (ans + zero - 1) << "\n";
        }
        else
        {
            cout << (digit - 1) << "\n";
        }
    }
    return 0;
}

File: CodeForces Contest Solutions/Codeforces_Round_1016 _(Div. 3)/B_Expensive_Number.java
import java.util.*;
public class B_Expensive_Number {
    static Scanner sc = new Scanner(System.in);
    
    public static void main(String[] args) throws Exception {
        int test = sc.nextInt();
        while (test-- > 0) {
            solve();
        }
        sc.close();
    }

    private static void solve() {
        char[] s = sc.next().toCharArray();

        long ans = 0, zero = 0;
        boolean haveZero = false, nonZero = false;

        for(int i=s.length-1; i>=0; i--) {
            int ld = s[i] - '0';

            if(ld == 0) haveZero = true;

            if(ld == 0 && !nonZero) {
                zero++;
            } else if(ld != 0) {
                ans++;
                nonZero = true;
            }
        }

        if(haveZero) {
            System.out.println(ans - 1 + zero);
        } else {
            System.out.println(s.length - 1);
        }

    }
}